imap_processing/codice/codice_l2.py

```diff
@@ -866,6 +957,162 @@ def process_hi_sectored(dependencies: ProcessingInputCollection) -> xr.Dataset:
     return l2_dataset
 
 
+def process_lo_direct_events(dependencies: ProcessingInputCollection) -> xr.Dataset:
+    """
+    Process the lo-direct-events L1A dataset to convert variables to physical units.
+
+    See section 11.2.1 of the CoDICE algorithm document for details.
+
+    Parameters
+    ----------
+    dependencies : ProcessingInputCollection
+        The collection of processing input files.
+
+    Returns
+    -------
+    xarray.Dataset
+        The updated L2 dataset with variables converted to physical units.
+    """
+    file_path = dependencies.get_file_paths(descriptor="lo-direct-events")[0]
+    l1a_dataset = load_cdf(file_path)
+
+    # Update global CDF attributes
+    cdf_attrs = ImapCdfAttributes()
+    cdf_attrs.add_instrument_global_attrs("codice")
+    cdf_attrs.add_instrument_variable_attrs("codice", "l2-lo-direct-events")
+    energy_table, energy_bins = get_lo_de_energy_luts(dependencies)
+    # Convert from position to elevation angle in degrees relative to the spacecraft
+    # axis
+    l2_dataset = l1a_dataset.copy(deep=True)
+    # Create a new coordinate for elevation_angle based on inst_az
+    pos_to_els = (
+        LO_POSITION_TO_ELEVATION_ANGLE["sw"] | LO_POSITION_TO_ELEVATION_ANGLE["nsw"]
+    )
+    elevation_angle_shape = l2_dataset["position"].shape
+    elevation_angle = np.array(
+        [pos_to_els.get(pos, np.nan) for pos in l2_dataset["position"].values.flat]
+    ).reshape(elevation_angle_shape)
+    l2_dataset["elevation_angle"] = (
+        l2_dataset["position"].dims,
+        elevation_angle.astype(np.float32),
+    )
+    # Convert spin_sector to spin_angle in degrees
+    # Use equation from section 11.2.2 of algorithm document
+    # Shift all spin sectors for all positions 13 - 24 adding 12 and mod 24
+    l2_dataset["spin_sector"] = xr.where(
+        (l2_dataset["position"] >= 13) & (l2_dataset["position"] <= 24),
+        (l2_dataset["spin_sector"] + 12) % 24,
+        l2_dataset["spin_sector"],
+    )
+    l2_dataset["spin_angle"] = xr.DataArray(
+        data=l2_dataset["spin_sector"].data * 15.0 + 7.5,
+        dims=l2_dataset["spin_sector"].dims,
+    ).astype(np.float32)
```

Comment:
I think you can do this directly without needing to create an DataArray yourself.

```suggestion
    l2_dataset["spin_angle"] = l2_dataset["spin_sector"].astype(np.float32) * 15.0 + 7.5
```